Sources/SnapshotTesting/Snapshotting/NSImage.swift

```diff
@@ -57,61 +59,46 @@ private func NSImagePNGRepresentation(_ image: NSImage) -> Data? {
+  }
+  guard let oldContext = context(for: oldCgImage), let oldData = oldContext.data else {
+    return "Reference image's data could not be loaded."
+  }
+  guard let newContext = context(for: newCgImage), let newData = newContext.data else {
+    return "Newly-taken snapshot's data could not be loaded."
+  }
   let byteCount = oldContext.height * oldContext.bytesPerRow
-  if memcmp(oldData, newData, byteCount) == 0 { return true }
-  let newer = NSImage(data: NSImagePNGRepresentation(new)!)!
-  guard let newerCgImage = newer.cgImage(forProposedRect: nil, context: nil, hints: nil) else { return false }
-  guard let newerContext = context(for: newerCgImage) else { return false }
-  guard let newerData = newerContext.data else { return false }
-  if memcmp(oldData, newerData, byteCount) == 0 { return true }
-  if precision >= 1, perceptualPrecision >= 1 { return false }
+  if memcmp(oldData, newData, byteCount) == 0 { return nil }
+  guard
+    let pngData = NSImagePNGRepresentation(new),
+    let newerCgImage = NSImage(data: pngData)?.cgImage(forProposedRect: nil, context: nil, hints: nil),
+    let newerContext = context(for: newerCgImage),
+    let newerData = newerContext.data
+  else {
+    return "Newly-taken snapshot's data could not be loaded."
+  }
+  if memcmp(oldData, newerData, byteCount) == 0 { return nil }
+  if precision >= 1, perceptualPrecision >= 1 {
+    return "Newly-taken snapshot does not match reference."
+  }
   if perceptualPrecision < 1, #available(macOS 10.13, *) {
-    let deltaFilter = CIFilter(
-      name: "CILabDeltaE",
-      parameters: [
-        kCIInputImageKey: CIImage(cgImage: newCgImage),
-        "inputImage2": CIImage(cgImage: oldCgImage)
-      ]
-    )
-    guard let deltaOutputImage = deltaFilter?.outputImage else { return false }
-    let extent = CGRect(x: 0, y: 0, width: oldCgImage.width, height: oldCgImage.height)
-    guard 
-      let thresholdOutputImage = try? ThresholdImageProcessorKernel.apply(
-        withExtent: extent,
-        inputs: [deltaOutputImage],
-        arguments: [ThresholdImageProcessorKernel.inputThresholdKey: (1 - perceptualPrecision) * 100]
-      )
-    else { return false }
-    let averageFilter = CIFilter(
-      name: "CIAreaAverage",
-      parameters: [
-        kCIInputImageKey: thresholdOutputImage,
-        kCIInputExtentKey: extent
-      ]
-    )
-    guard let averageOutputImage = averageFilter?.outputImage else { return false }
-    var averagePixel: Float = 0
-    CIContext(options: [.workingColorSpace: NSNull(), .outputColorSpace: NSNull()]).render(
-      averageOutputImage,
-      toBitmap: &averagePixel,
-      rowBytes: MemoryLayout<Float>.size,
-      bounds: CGRect(x: 0, y: 0, width: 1, height: 1),
-      format: .Rf,
-      colorSpace: nil
+    return perceptuallyCompare(
+      CIImage(cgImage: oldCgImage),
+      CIImage(cgImage: newCgImage),
+      pixelPrecision: precision,
+      perceptualPrecision: perceptualPrecision
```

Comment:
The CoreImage operations were [extracted into a separate method](https://github.com/pointfreeco/swift-snapshot-testing/pull/638/files#diff-3e1681b0623d05afda1993aa39ee2ae8641918d2e1df61acf80b1a367dada06fR174-R218) since they were common across `UIImage.swift` and `NSImage.swift`